### PR TITLE
Sync up with changes from voting branch

### PR DESCRIPTION
**Pull request by ManfredKarrer**
There have been several changes in the voting branch (and api-integration) which I want to get into master to avoid that those branches run out of sync as well as to apply the improvements.
The main changes are:
- The handling of persisted data in the p2p library (https://github.com/bisq-network/bisq-p2p/issues/8)
- Improvements of the app setup to support API projects in headless mode as well as in mixed mode
- Use custom TxBroadcaster class for better error handling 

Beside that there are many smaller trivial improvements, fixes, refactoring and cleanups.

### CODE REVIEW

File: src/main/proto/pb.proto [hunk 1]
```diff
@@ -1009,16 +1009,24 @@ message PersistableEnvelope {
         UserPayload user_payload = 10;
         PaymentAccountList payment_account_list = 11;
 
-        CompensationRequestPayload compensation_request_payload = 12; // deprecated. Not used anymore.
```
Comment (sqrrm) [inline src/main/proto/pb.proto:1012]: Doesn't this break backwards compatibility?

Comment (ManfredKarrer) [reply src/main/proto/pb.proto:1012]: It is not used by Bisq users, so should have no effect. At least with my tests between old and new version I did not see any issue.

Comment (sqrrm) [reply src/main/proto/pb.proto:1012]: Ah, that's true, this is not out yet and more reorg here is to be expected.

**Review by sqrrm — APPROVED** [review]
utACK

**Comment by ripcurlx** [comment]
ACK - see https://github.com/bisq-network/bisq-desktop/pull/1601#issuecomment-403486954

**Pull request merged**